src/components/views/settings/tabs/user/VoiceUserSettingsTab.tsx

```diff
@@ -43,6 +45,7 @@ const getDefaultDevice = (devices: Array<Partial<MediaDeviceInfo>>) => {
 
 interface IState extends Record<MediaDeviceKindEnum, string> {
     mediaDevices: IMediaDevices;
+    audioSettings: AudioSettings;
```

Comment:
I wonder if it would be preferable to have this split into the individual settings

Reply:
I'll try to split it into separate settings.

I had a hard time keeping the application state together:
- the persisted device settings
- the state in the React component by which rendering is done
- the state of `MediaDeviceHandler`, which is propagated to `matrix-js-sdk`'s media handler

If my understanding is correct, using some kind of a state management solution that is well-integrated with React (MobX, Recoil, Redux) would be a good resolution to my problem.